packages/core/src/modules/dids/domain/__tests__/DidDocument.test.ts

```diff
@@ -152,100 +152,11 @@ describe('Did | DidDocument', () => {
     expect(keyAgreement[1]).toBeInstanceOf(VerificationMethod)
   })
 
-  it('validation should throw an error if the did document is invalid', async () => {
-    const didDocument = JsonTransformer.fromJSON(didExample456Invalid, DidDocument)
-
+  it('validation should throw an error if the did document is invalid', () => {
     try {
-      await MessageValidator.validate(didDocument)
+      JsonTransformer.fromJSON(didExample456Invalid, DidDocument)
     } catch (error) {
-      expect(error).toMatchObject([
```

Comment:
we're losing some important test data here. Can we add an `errors` property to the class validation error class and match whether the errors match this object? I'd like to keep the test for specific validation errors in here

Reply:
weird this is losing all this data. If you solve this one error, will it give a new error? (i.e. it will deliver them one by one instead)